src/net/session.c

```diff
@@ -188,6 +189,7 @@ int8_t _zp_start_read_task(_z_session_t *zn) {
                 zn->_tp._transport._multicast._read_task_running = false;
                 ret = _Z_ERR_SYSTEM_TASK_FAILED;
                 z_free(task);
+                zn->_tp._transport._multicast._read_task = NULL;
             }
```

Comment:
can you also change the previous line `z_free(task)` --> `z_free(zn->_tp._transport._unicast._read_task)`.